core/src/block_sync.rs

```diff
@@ -8,75 +8,52 @@ use std::{fmt::Debug, sync::Arc, time::Duration};
 
 use iroha_actor::{broker::*, prelude::*, Context};
 use iroha_config::block_sync::Configuration;
-use iroha_crypto::SignatureOf;
+use iroha_crypto::{SignatureOf, *};
 use iroha_data_model::prelude::*;
 use iroha_logger::prelude::*;
-use rand::{prelude::SliceRandom, SeedableRng};
+use iroha_macro::*;
+use iroha_p2p::Post;
+use iroha_version::prelude::*;
+use parity_scale_codec::{Decode, Encode};
 
-use self::message::{Message, *};
 use crate::{
     prelude::*,
-    sumeragi::{
-        message::{CommitBlock, GetNetworkTopology},
-        network_topology::Role,
-        SumeragiTrait,
-    },
-    VersionedCommittedBlock,
+    sumeragi::{network_topology::Role, Sumeragi},
+    NetworkMessage, VersionedCommittedBlock,
 };
 
-/// The state of `BlockSynchronizer`.
-#[derive(Clone, Debug)]
-enum State {
-    /// Not synchronizing now.
-    Idle,
-    /// Synchronization is in progress: validating and committing blocks.
-    /// Contains a vector of blocks left to commit and an id of the peer from which the blocks were requested.
-    InProgress(Vec<VersionedCommittedBlock>, PeerId),
-}
-
 /// Structure responsible for block synchronization between peers.
 #[derive(Debug)]
-pub struct BlockSynchronizer<S: SumeragiTrait> {
-    wsv: Arc<WorldStateView>,
-    sumeragi: AlwaysAddr<S>,
+pub struct BlockSynchronizer {
+    sumeragi: Arc<Sumeragi>,
     peer_id: PeerId,
-    state: State,
     gossip_period: Duration,
     block_batch_size: u32,
     broker: Broker,
     actor_channel_capacity: u32,
 }
 
 /// Block synchronizer
-pub trait BlockSynchronizerTrait: Actor + Handler<ContinueSync> + Handler<Message> {
-    /// Requires sumeragi for sending direct messages to it
-    type Sumeragi: SumeragiTrait;
-
+pub trait BlockSynchronizerTrait: Actor + Handler<Message> {
```

Comment:
What is the purpose of this trait? 
I understand that this is not your change but trait seems to be redundant here.

Reply:
Yes I think we should remove this one as well.

Reply:
Removed